In [1]:
import ast
import pandas as pd

reviews = []

# Abre el archivo 'australian_user_reviews.json' en modo de lectura
with open('DataSet/australian_user_reviews.json', 'r', encoding='utf-8') as f:
    # Itera sobre cada línea del archivo
    for line in f.readlines():
        # Convierte la línea (que es una cadena de texto) en una estructura de datos de Python usando 'ast.literal_eval'
        reviews.append(ast.literal_eval(line))

# Crea un DataFrame de pandas a partir de la lista de reseñas
df_ur = pd.DataFrame(reviews)

df_ur



user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1                js41637               http://steamcommunity.com/id/js41637   
2              evcentric             http://steamcommunity.com/id/evcentric   
3                  doctr                 http://steamcommunity.com/id/doctr   
4              maplemage             http://steamcommunity.com/id/maplemage   
...                  ...                                                ...   
25794  76561198306599751  http://steamcommunity.com/profiles/76561198306...   
25795           Ghoustik              http://steamcommunity.com/id/Ghoustik   
25796  76561198310819422  http://steamcommunity.com/profiles/76561198310...   
25797  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
25798        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                                 reviews  
0      [{'funny': '', 'posted': 'Posted November 5, 2...  
1      [{'funny': '', 'posted': 'Posted June 24, 2014...  
2      [{'funny': '', 'posted': 'Posted February 3.',...  
3      [{'funny': '', 'posted': 'Posted October 14, 2...  
4      [{'funny': '3 people found this review funny',...  
...                                                  ...  
25794  [{'funny': '', 'posted': 'Posted May 31.', 'la...  
25795  [{'funny': '', 'posted': 'Posted June 17.', 'l...  
25796  [{'funny': '1 person found this review funny',...  
25797  [{'funny': '', 'posted': 'Posted July 21.', 'l...  
25798  [{'funny': '1 person found this review funny',...  

[25799 rows x 3 columns]

In [2]:
# Ver valores nulos en df_ur
print("Valores nulos en df_ur:")
print(df_ur.isnull().sum())

Valores nulos en df_ur:
user_id     0
user_url    0
reviews     0
dtype: int64


In [3]:
# Obtener información general del DataFrame

df_ur.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25799 entries, 0 to 25798
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   user_id   25799 non-null  object
 1   user_url  25799 non-null  object
 2   reviews   25799 non-null  object
dtypes: object(3)
memory usage: 604.8+ KB


## Columna user_url

In [4]:
# Imprimir la columna 'user_url'
print(df_ur['user_url'])

0        http://steamcommunity.com/profiles/76561197970...
1                     http://steamcommunity.com/id/js41637
2                   http://steamcommunity.com/id/evcentric
3                       http://steamcommunity.com/id/doctr
4                   http://steamcommunity.com/id/maplemage
                               ...                        
25794    http://steamcommunity.com/profiles/76561198306...
25795                http://steamcommunity.com/id/Ghoustik
25796    http://steamcommunity.com/profiles/76561198310...
25797    http://steamcommunity.com/profiles/76561198312...
25798             http://steamcommunity.com/id/LydiaMorley
Name: user_url, Length: 25799, dtype: object


Eliminamos la columna user_url pues no es útil para el proyecto

In [5]:
# Eliminar la columna 'user_url'
df_ur.drop('user_url', axis=1, inplace=True)

## Columna user_id

In [6]:
# Imprimir la columna 'user_id'
print(df_ur['user_id'])

0        76561197970982479
1                  js41637
2                evcentric
3                    doctr
4                maplemage
               ...        
25794    76561198306599751
25795             Ghoustik
25796    76561198310819422
25797    76561198312638244
25798          LydiaMorley
Name: user_id, Length: 25799, dtype: object


In [7]:
# Valores únicos en la columna 'user_id'
unique_itemn_count = df_ur['user_id'].nunique()
print("Cantidad de user_id únicos:", unique_itemn_count)

Cantidad de user_id únicos: 25485


## Columna reviews

In [8]:
# Imprimir la columna 'reviews'
print(df_ur['reviews'])

0        [{'funny': '', 'posted': 'Posted November 5, 2...
1        [{'funny': '', 'posted': 'Posted June 24, 2014...
2        [{'funny': '', 'posted': 'Posted February 3.',...
3        [{'funny': '', 'posted': 'Posted October 14, 2...
4        [{'funny': '3 people found this review funny',...
                               ...                        
25794    [{'funny': '', 'posted': 'Posted May 31.', 'la...
25795    [{'funny': '', 'posted': 'Posted June 17.', 'l...
25796    [{'funny': '1 person found this review funny',...
25797    [{'funny': '', 'posted': 'Posted July 21.', 'l...
25798    [{'funny': '1 person found this review funny',...
Name: reviews, Length: 25799, dtype: object


In [9]:
# Expande la columna 'reviews' que contiene listas de reseñas en filas individuales
dfreviewsOpen = df_ur.explode('reviews') 
# Combina el DataFrame original (sin columna 'reviews') con las columnas extraídas de la columna 'reviews'. Convierte cada diccionario en la columna 'reviews' en una serie.
df_ur = pd.concat([dfreviewsOpen.drop(['reviews'], axis=1), dfreviewsOpen['reviews'].apply(pd.Series)], axis=1)

In [10]:
df_ur

user_id                             funny  \
0      76561197970982479                                     
0      76561197970982479                                     
0      76561197970982479                                     
1                js41637                                     
1                js41637                                     
...                  ...                               ...   
25797  76561198312638244                                     
25797  76561198312638244                                     
25798        LydiaMorley  1 person found this review funny   
25798        LydiaMorley                                     
25798        LydiaMorley                                     

                          posted last_edited item_id  \
0       Posted November 5, 2011.                1250   
0          Posted July 15, 2011.               22200   
0         Posted April 21, 2011.               43110   
1          Posted June 24, 2014.              251610   
1      Posted September 8, 2013.              227300   
...                          ...         ...     ...   
25797            Posted July 10.                  70   
25797             Posted July 8.              362890   
25798             Posted July 3.              273110   
25798            Posted July 20.                 730   
25798             Posted July 2.                 440   

                                               helpful recommend  \
0                                       No ratings yet      True   
0                                       No ratings yet      True   
0                                       No ratings yet      True   
1      15 of 20 people (75%) found this review helpful      True   
1         0 of 1 people (0%) found this review helpful      True   
...                                                ...       ...   
25797                                   No ratings yet      True   
25797                                   No ratings yet      True   
25798    1 of 2 people (50%) found this review helpful      True   
25798                                   No ratings yet      True   
25798                                   No ratings yet      True   

                                                  review   0  
0      Simple yet with great replayability. In my opi... NaN  
0                   It's unique and worth a playthrough. NaN  
0      Great atmosphere. The gunplay can be a bit chu... NaN  
1      I know what you think when you see this title ... NaN  
1      For a simple (it's actually not all that simpl... NaN  
...                                                  ...  ..  
25797  a must have classic from steam definitely wort... NaN  
25797  this game is a perfect remake of the original ... NaN  
25798  had so much fun plaing this and collecting res... NaN  
25798                                                 :D NaN  
25798                                     so much fun :D NaN  

[59333 rows x 9 columns]

In [11]:
# Eliminar columna 0
df_ur.drop(0, axis=1, inplace=True)

## Columna funny

In [12]:
# Imprimir la columna 'funny'
print(df_ur['funny'])

0                                        
0                                        
0                                        
1                                        
1                                        
                       ...               
25797                                    
25797                                    
25798    1 person found this review funny
25798                                    
25798                                    
Name: funny, Length: 59333, dtype: object


El contenido de la columna no tiene relevancia para el proyecto. Se procede a eliminar la columna "funny"

In [13]:
# Eliminar la columna 'funny'
df_ur.drop('funny', axis=1, inplace=True)

## Columna posted

In [14]:
# Imprimir la columna 'posted'
print(df_ur['posted'])

0         Posted November 5, 2011.
0            Posted July 15, 2011.
0           Posted April 21, 2011.
1            Posted June 24, 2014.
1        Posted September 8, 2013.
                   ...            
25797              Posted July 10.
25797               Posted July 8.
25798               Posted July 3.
25798              Posted July 20.
25798               Posted July 2.
Name: posted, Length: 59333, dtype: object


In [15]:
# Contar valores nulos en la columna 'posted'
nulos_posted = df_ur['posted'].isnull().sum()
nulos_posted

28

Son 28 filas nulas de un total de 59333. Al ser estadísticamente irrelevante, se procede a eliminar los nulos.

In [16]:
# Eliminar filas con valores nulos en la columna 'posted'
df_ur = df_ur.dropna(subset=['posted'])

In [17]:
# Obtener una muestra aleatoria de 20 datos de la columna 'posted'
random_sample = df_ur['posted'].sample(20)

# Mostrar la muestra aleatoria
print(random_sample)

13936       Posted August 12, 2014.
20988       Posted January 2, 2014.
20781      Posted January 18, 2015.
13296       Posted August 16, 2013.
20845     Posted September 8, 2014.
4197       Posted October 29, 2014.
25599    Posted September 13, 2014.
1267          Posted July 25, 2015.
2878           Posted June 6, 2014.
14697         Posted June 26, 2014.
17525         Posted April 7, 2013.
6243        Posted January 2, 2014.
22812        Posted April 11, 2015.
7059       Posted November 6, 2015.
23055      Posted January 26, 2014.
7115      Posted December 10, 2013.
22044     Posted November 20, 2013.
12522                Posted July 6.
20492         Posted March 9, 2015.
9681      Posted December 23, 2015.
Name: posted, dtype: object


La palabra "Posted" presente en las filas de la columna, nos impide convertir la misma a tipo de dato date o int, por lo que debemos eliminarla.

In [18]:
# Eliminar la palabra 'Posted' de la columna 'posted'
df_ur['posted'] = df_ur['posted'].str.replace('Posted ', '')

C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\3004603827.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur['posted'] = df_ur['posted'].str.replace('Posted ', '')


In [19]:
# Obtener una muestra aleatoria de 20 datos de la columna 'posted'
random_sample = df_ur['posted'].sample(20)

# Mostrar la muestra aleatoria
print(random_sample)

9361     September 30, 2015.
23391       January 2, 2014.
97                   July 8.
16120               June 27.
7197     September 27, 2015.
18269      January 19, 2014.
8893         March 12, 2015.
5796          June 14, 2014.
20169      December 6, 2014.
418           April 2, 2015.
807       November 18, 2014.
18482      November 5, 2015.
20713         April 5, 2015.
22057      December 5, 2014.
21215        April 30, 2014.
20531        March 15, 2013.
11476               June 27.
16590          June 1, 2014.
12215         June 20, 2014.
14023          May 11, 2014.
Name: posted, dtype: object


In [20]:
import re

# Extraer el año y convertir el resultado a un tipo de dato numérico
# Obtenemos los años únicos no nulos directamente
unique_years = df_ur['posted'].str.extract(r'(\d{4})')[0].dropna().unique()

print(unique_years)


['2011' '2014' '2013' '2015' '2012' '2010']


Se observa que hay valores con año y otros sin año asignado. Se asume que los que no tienen año específicado es porque son del año en curso, es decir, de cuando se tomó esta muestra. Se puede inferir que el año en que se tomó la muestra es el año 2016, pues los años asignados en las filas llegan hasta el 2015.



In [21]:
# Crear la nueva columna 'posted_year_' y transferir los datos de 'posted'
df_ur['posted_year_'] = pd.to_datetime(df_ur['posted'], errors='coerce')

C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\2188426456.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur['posted_year_'] = pd.to_datetime(df_ur['posted'], errors='coerce')


In [22]:
# Contar los valores nulos en 'posted_year_'
nulos_posted_year = df_ur['posted_year_'].isnull().sum()

# Mostrar el resultado
print("Número de valores nulos en 'posted_year_':", nulos_posted_year)

Número de valores nulos en 'posted_year_': 10119


In [23]:
# Imprimir la columna 'posted_year_'
print(df_ur['posted_year_'])

0       2011-11-05
0       2011-07-15
0       2011-04-21
1       2014-06-24
1       2013-09-08
           ...    
25797          NaT
25797          NaT
25798          NaT
25798          NaT
25798          NaT
Name: posted_year_, Length: 59305, dtype: datetime64[ns]


Se advierte que los valores nulos corresponden a aquellos datos que no tienen año asignado. Como se mencionó más arriba, estos datos corresponden al año en que se tomó la muestra: 2016. 

In [24]:
# Extraer solo el año de la columna 'posted_year_'
df_ur['posted_year'] = df_ur['posted_year_'].dt.year

# Imprimir la columna 'posted_year'
print(df_ur['posted_year'])

0        2011.0
0        2011.0
0        2011.0
1        2014.0
1        2013.0
          ...  
25797       NaN
25797       NaN
25798       NaN
25798       NaN
25798       NaN
Name: posted_year, Length: 59305, dtype: float64


C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\2634345063.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur['posted_year'] = df_ur['posted_year_'].dt.year


In [25]:
# Rellenar los valores nulos en 'posted_year' con 2016
df_ur['posted_year'] = df_ur['posted_year_'].fillna('2016')

# Verificar el resultado
print("Número de valores nulos en 'posted_year':", df_ur['posted_year'].isnull().sum())

Número de valores nulos en 'posted_year': 0


C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\2697398903.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur['posted_year'] = df_ur['posted_year_'].fillna('2016')


In [26]:
# Convertir la columna 'posted_year' a entero
df_ur['posted_year'] = df_ur['posted_year'].dt.year.astype(int)

# Verificar el resultado
print("Tipo de datos de 'posted_year':", df_ur['posted_year'].dtype)

Tipo de datos de 'posted_year': int32


C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\4229553040.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur['posted_year'] = df_ur['posted_year'].dt.year.astype(int)


In [27]:
# Eliminar las columnas 'posted_year_' y 'posted'
df_ur.drop(['posted_year_', 'posted'], axis=1, inplace=True)

C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\2382188425.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur.drop(['posted_year_', 'posted'], axis=1, inplace=True)


In [28]:
# Obtener valores únicos de la columna 'posted_year'
unique_posted_year = df_ur['posted_year'].unique()

# Mostrar los valores únicos
print("Valores únicos en 'posted_year':", unique_posted_year)


Valores únicos en 'posted_year': [2011 2014 2013 2016 2015 2012 2010]


## Columna last_edited

In [29]:
# Imprimir la columna 'last_edited'
print(df_ur['last_edited'])

0         
0         
0         
1         
1         
        ..
25797     
25797     
25798     
25798     
25798     
Name: last_edited, Length: 59305, dtype: object


In [30]:
# Obtener una muestra aleatoria de 20 datos de la columna 'last_edited'
random_sample = df_ur['last_edited'].sample(20)

# Mostrar la muestra aleatoria
print(random_sample)

11602                                  
6434                                   
792                                    
1505                                   
14411                                  
25246                                  
24106                                  
12104                                  
5023                                   
16062                                  
2055                                   
1768                                   
15080                                  
12479                                  
3414                                   
5393                                   
15760    Last edited November 18, 2015.
16712                                  
2840            Last edited February 7.
22880                                  
Name: last_edited, dtype: object


La columna "last_edited" cuenta con muchos valores nulos y, además, su información no es útil a efectos del presente proyecto. Se procede a eliminar la columna.

In [31]:
# Eliminar la columna 'last_edited'
df_ur.drop('last_edited', axis=1, inplace=True)

C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\4236713998.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur.drop('last_edited', axis=1, inplace=True)


## Columna item_id

In [32]:
# Imprimir la columna 'item_id'
print(df_ur['item_id'])

0          1250
0         22200
0         43110
1        251610
1        227300
          ...  
25797        70
25797    362890
25798    273110
25798       730
25798       440
Name: item_id, Length: 59305, dtype: object


In [33]:
# Sumar nulos en la columna 'item_id'
print("Cantidad de valores nulos en 'item_id':", df_ur['item_id'].isnull().sum())

Cantidad de valores nulos en 'item_id': 0


In [34]:
# Convertir la columna 'item_id' a entero
df_ur['item_id'] = pd.to_numeric(df_ur['item_id'], errors='coerce', downcast='integer')

C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\2316444153.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur['item_id'] = pd.to_numeric(df_ur['item_id'], errors='coerce', downcast='integer')


## Columna helpful

In [35]:
# Imprimir la columna 'helpful'
print(df_ur['helpful'])

0                                         No ratings yet
0                                         No ratings yet
0                                         No ratings yet
1        15 of 20 people (75%) found this review helpful
1           0 of 1 people (0%) found this review helpful
                              ...                       
25797                                     No ratings yet
25797                                     No ratings yet
25798      1 of 2 people (50%) found this review helpful
25798                                     No ratings yet
25798                                     No ratings yet
Name: helpful, Length: 59305, dtype: object


No es relevante la información de esta columna para el propósito del presente proyecto.

In [36]:
# Eliminar la columna 'helpful'
df_ur.drop('helpful', axis=1, inplace=True)


C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\2926190794.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur.drop('helpful', axis=1, inplace=True)


## Columna recommend

In [37]:
# Imprimir la columna 'recommend'
print(df_ur['recommend'])

0        True
0        True
0        True
1        True
1        True
         ... 
25797    True
25797    True
25798    True
25798    True
25798    True
Name: recommend, Length: 59305, dtype: object


In [38]:
# Convertir la columna 'recommend' a tipo booleano
df_ur['recommend'] = df_ur['recommend'].astype(bool)

C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\3172642279.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur['recommend'] = df_ur['recommend'].astype(bool)


## Columna review y análisis de sentimiento

In [39]:
# Imprimir la columna 'review'
print(df_ur['review'])


0        Simple yet with great replayability. In my opi...
0                     It's unique and worth a playthrough.
0        Great atmosphere. The gunplay can be a bit chu...
1        I know what you think when you see this title ...
1        For a simple (it's actually not all that simpl...
                               ...                        
25797    a must have classic from steam definitely wort...
25797    this game is a perfect remake of the original ...
25798    had so much fun plaing this and collecting res...
25798                                                   :D
25798                                       so much fun :D
Name: review, Length: 59305, dtype: object


Procedemos con el análisis de sentimiento que se nos solicita.

In [40]:
from textblob import TextBlob

# Función para realizar análisis de sentimiento
def analyze_sentiment(review):
    if pd.isnull(review):  # Si la reseña es nula, devolver 1 (neutral)
        return 1
    else:
        analysis = TextBlob(review)
        # Asignar valores de acuerdo con la polaridad del análisis de sentimiento
        if analysis.sentiment.polarity < 0:
            return 0  # Sentimiento negativo
        elif analysis.sentiment.polarity == 0:
            return 1  # Sentimiento neutro
        else:
            return 2  # Sentimiento positivo

# Aplicar la función a la columna 'review' y crear la nueva columna 'sentiment_analysis'
df_ur['sentiment_analysis'] = df_ur['review'].apply(analyze_sentiment)

# Mostrar información después del proceso
print(df_ur[['review', 'sentiment_analysis']].head())

                                              review  sentiment_analysis
0  Simple yet with great replayability. In my opi...                   2
0               It's unique and worth a playthrough.                   2
0  Great atmosphere. The gunplay can be a bit chu...                   2
1  I know what you think when you see this title ...                   2
1  For a simple (it's actually not all that simpl...                   0


C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\3435470671.py:18: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur['sentiment_analysis'] = df_ur['review'].apply(analyze_sentiment)


In [41]:
# Eliminar la columna 'review'
df_ur.drop(['review'], axis=1, inplace=True)

C:\Users\nicol\AppData\Local\Temp\ipykernel_22060\2841975229.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ur.drop(['review'], axis=1, inplace=True)


Revisamos información final del DataFrame, reiniciamos el índice y guardamos la información en formato ".csv".

In [42]:
# Mostrar información general y valores nulos del DataFrame
df_ur.info()

# Contar los valores nulos en cada columna
nulos_por_columna = df_ur.isnull().sum()

# Mostrar la cantidad de valores nulos por columna
print("Valores nulos por columna:")
print(nulos_por_columna)


<class 'pandas.core.frame.DataFrame'>
Index: 59305 entries, 0 to 25798
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   user_id             59305 non-null  object
 1   item_id             59305 non-null  int32 
 2   recommend           59305 non-null  bool  
 3   posted_year         59305 non-null  int32 
 4   sentiment_analysis  59305 non-null  int64 
dtypes: bool(1), int32(2), int64(1), object(1)
memory usage: 1.9+ MB
Valores nulos por columna:
user_id               0
item_id               0
recommend             0
posted_year           0
sentiment_analysis    0
dtype: int64


In [43]:
# Reiniciar índice
df_ur = df_ur.reset_index(drop=True)

In [44]:

# Rutas de los archivos CSV y Parquet
# csv_path = 'DataSet/clean_df_ur.csv'
parquet_path = 'DataSet/clean_df_ur.parquet'

# Guardar el DataFrame como un archivo CSV
# df_ur.to_csv(csv_path, index=False)

# Guardar el DataFrame como un archivo Parquet
df_ur.to_parquet(parquet_path, index=False)
print(f'El archivo Parquet {parquet_path} ha sido creado.')



El archivo Parquet DataSet/clean_df_ur.parquet ha sido creado.
